nengo/utils/threading.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import absolute_import
+
+import collections
+import threading
+
+
+class ThreadLocalContext(threading.local, collections.Sequence):
+    def __init__(self):
+        super(ThreadLocalContext, self).__init__()
+        self._context = []
+
+    def __len__(self):
+        return len(self._context)
+
+    def __getitem__(self, i):
+        return self._context[i]
+
+    def append(self, item):
+        self._context.append(item)
+
+    def pop(self):
+        return self._context.pop()
+
+    def clear(self):
+        self._context = []
```

Comment:
Probably not a big deal, but I usually prefer `del self._context[:]` just to keep using the same list object.